src/freenas/usr/local/bin/ixdiagnose

```diff
@@ -80,14 +85,17 @@ truncate_files ()
 	done
 }
 
-test -e /etc/rc.conf && . /etc/rc.conf
-[ "$dumpdir" = "" ] && dumpdir="$dumpdir"
+if [ "$is_linux" -eq 1 ]; then
```

Comment:
We'll be misunderstanding this as "if we are on linux"

Reply:
In the next line i have a comment clarifying that this section is freebsd specific, please advise if you think i should clarify this further somehow. Thank you

Reply:
We could create a small library to source in scripts, with something along the lines of:
```sh
case $(uname -s) in
FreeBSD)
	is_freebsd=true
	is_linux=false
	;;
Linux)
	is_freebsd=false
	is_linux=true
	;;
esac

is_freebsd() {
	$is_freebsd
}

is_linux() {
	$is_linux
}
```

That will give us easy to use functions:
```sh
source /etc/system # or whatever the lib gets called

if is_freebsd; then
	do_freebsd_things
elif is_linux; then
	do_linux_things
fi
```

This is what has been done in the ZFS test suite to support multiple platforms.

Reply:
Thank you @freqlabs, i have updated the code base to use such a script.